git/util.py

```diff
@@ -395,17 +396,8 @@ def _parse_progress_line(self, line):
         for sline in sub_lines:
             # find escape characters and cut them away - regex will not work with
             # them as they are non-ascii. As git might expect a tty, it will send them
-            last_valid_index = None
-            for i, c in enumerate(reversed(sline)):
-                if ord(c) < 32:
-                    # its a slice index
-                    last_valid_index = -i - 1
-                # END character was non-ascii
-            # END for each character in sline
-            if last_valid_index is not None:
-                sline = sline[:last_valid_index]
-            # END cut away invalid part
-            sline = sline.rstrip()
+            sline = self.re_ansi_escape.sub('', sline)
+            sline.strip()
```

Comment:
`strip()` will not mutate the string. Thus it should be `sline = sline.strip()`.
If it is important, do you think it's possible to have a test for it? Otherwise, maybe that line is not needed at all.

Reply:
I just saw that the line stripping was done previously as well, let's just re-add it then as in `sline = sline.rstrip()`. Apparently there is no test that would need it, and it's tech-debt I wouldn't ask you to pay either 😅.

Reply:
Absolutely, good catch!